ci/Jenkinsfile-mac-integration

```diff
@@ -215,31 +215,41 @@ def builders = [:]
 
 
 builders['mac-tests'] = testBuilder('mac', 'mac')({
-    stage ("Run dcos-cli tests") {
-        sh '''
-           rm -rf ~/.dcos; \
-           cp /etc/hosts hosts.local; \
-           grep -q "^.* dcos.snakeoil.mesosphere.com$" hosts.local && \
-           sed -iold "s/^.* dcos.snakeoil.mesosphere.com$/${DCOS_IP} dcos.snakeoil.mesosphere.com/" hosts.local || \
-           echo ${DCOS_IP} dcos.snakeoil.mesosphere.com >> hosts.local; \
-           sudo cp ./hosts.local /etc/hosts'''
-
-        dir('dcos-cli') {
-            sh 'make test'
-        }
+    try {
+        stage ("Run dcos-cli tests") {
+            sh '''
+               rm -rf ~/.dcos; \
+               cp /etc/hosts hosts.local; \
+               grep -q "^.* dcos.snakeoil.mesosphere.com$" hosts.local && \
+               sed -iold "s/^.* dcos.snakeoil.mesosphere.com$/${DCOS_IP} dcos.snakeoil.mesosphere.com/" hosts.local || \
+               echo ${DCOS_IP} dcos.snakeoil.mesosphere.com >> hosts.local; \
+               sudo cp ./hosts.local /etc/hosts'''
 
-        dir('dcos-cli/cli') {
+            dir('dcos-cli') {
+                sh 'make test'
+            }
+
+            dir('dcos-cli/cli') {
+                sh '''
+                   export PYTHONIOENCODING=utf-8; \
+                   make binary; \
+                   export CLI_TEST_SSH_USER=centos; \
+                   export CLI_TEST_MASTER_PROXY=true; \
+                   dist/dcos cluster setup dcos.snakeoil.mesosphere.com \
+                     --insecure --username=${DCOS_ADMIN_USERNAME} \
+                     --password-env=DCOS_ADMIN_PASSWORD; \
+                   dist/dcos config set core.reporting false; \
+                   dist/dcos config set core.timeout 5; \
+                   make binary'''
+            }
+        }
+    } finally {
+        stage ("Cleaning up \$DCOS_DIR and /etc/hosts") {
             sh '''
-               export PYTHONIOENCODING=utf-8; \
-               make binary; \
-               export CLI_TEST_SSH_USER=centos; \
-               export CLI_TEST_MASTER_PROXY=true; \
-               dist/dcos cluster setup dcos.snakeoil.mesosphere.com \
-                   --insecure --username=${DCOS_ADMIN_USERNAME} \
-                   --password-env=DCOS_ADMIN_PASSWORD; \
-               dist/dcos config set core.reporting false; \
-               dist/dcos config set core.timeout 5; \
-               make test-binary'''
+               cat /etc/hosts; \
+               sed -i "/dcos.snakeoil.mesosphere.com/d" /etc/hosts; \
```

Comment:
I don't think this will work. You can't modify `/etc/hosts` directly because you need `sudo` access to it, which we don't allow for the jenkins user. The only `sudo` command we allow for the jenkins user is `cp /etc/hosts hosts.local` because we explicitly added an exception for that in the `sudoers` file. That means you will have to copy `/etc/hosts` to a file called `hosts.local`, modify it there, and then copy it over `/etc/hosts` using the `cp /etc/hosts hosts.local`  command. This is the same pattern we follow above.

Reply:
thanks for the tip, I'll update it as being done above.